src/orion/serving/storage_resource.py

```diff
@@ -0,0 +1,232 @@
+"""
+Module responsible for storage import/export REST endpoints
+===========================================================
+
+Serves all the requests made to storage import/export REST endpoints.
+
+"""
+import json
+import logging
+import multiprocessing
+import os
+import traceback
+import uuid
+from datetime import datetime
+from queue import Empty
+
+import falcon
+from falcon import Request, Response
+
+from orion.core.io.database import DatabaseError
+from orion.core.worker.storage_backup import dump_database, load_database
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+def _gen_host_file(basename="dump"):
+    """Generate a temporary file where data could be saved.
+
+    Create an empty file without collision in working directory.
+    Return name of generated file.
+    """
+    index = 0
+    while True:
+        file_name = f"{basename}{index}.pkl"
+        try:
+            with open(file_name, "x"):
+                return file_name
+        except FileExistsError:
+            index += 1
+            continue
+
+
+class Notifications:
+    """Stream handler to collect messages in a shared queue.
+
+    Used instead of stdout in import progress to capture log messages.
+    """
+
+    def __init__(self):
+        """Initialize with a shared queue."""
+        self.queue = multiprocessing.Queue()
+
+    def write(self, buf: str):
+        """Write received data"""
+        for line in buf.rstrip().splitlines():
+            self.queue.put(line)
+
+    def flush(self):
+        """Placeholder to flush data"""
+
+
+class ImportTask:
+    """Wrapper to represent an import task. Used to monitor task progress.
+
+    Properties:
```

Comment:
Should be formatted as an Attribute section in numpy docstring style.

Reply:
Ok!